vuu-ui/packages/vuu-layout/src/layout-persistence/LayoutPersistenceManager.ts

```diff
@@ -0,0 +1,46 @@
+import { LayoutJSON } from "@finos/vuu-layout";
+import { LayoutMetadata } from "@finos/vuu-shell";
+
+export interface LayoutPersistenceManager {
+  /**
+   * Saves a new layout
+   *
+   * @param metadata - Metadata about the layout to be saved
+   * @param layout   - Full JSON representation of the layout to be saved
+   *
+   * @returns ID assigned to the saved layout
+   */
+  createLayout: (metadata: Omit<LayoutMetadata, "id">, layout: LayoutJSON) => string;
+
+  /**
+   * Overwrites an existing layout with a new one
+   *
+   * @param id       - Unique identifier of the existing layout to be updated
+   * @param metadata - Metadata describing the new layout to overwrite with
+   * @param layout   - Full JSON representation of the new layout to overwrite with
+   */
+  updateLayout: (id: string, metadata: Omit<LayoutMetadata, "id">, layout: LayoutJSON) => void;
```

Comment:
on update I think we can just say metadata: LayoutMetadata rather than Omit<...> as we already have an id here

Reply:
I still prefer it this way, because it makes it clear that they're being used for different purposes: the ID is existing information used for retrieval, whereas the metadata is new information to be saved. It also makes the documentation a lot clearer, avoiding the need for a lengthy explanation against `@param metadata`.